### PR TITLE
docs: updates automatics reloading behaviour.

### PR DESCRIPTION
**Pull request by nilekhc**
Fixes https://github.com/kubernetes/kubernetes/issues/122563

/cc @enj @aramase

### CODE REVIEW

**Review by aramase — CHANGES_REQUESTED** [review]
```bash
--encryption-provider-config-automatic-reload boolean argument determines if the file set by --encryption-provider-config should be automatically reloaded if the disk contents change. This enables key rotation without API server restarts.
```
in https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/#encrypting-your-data-with-the-kms-provider needs to be updated as well.

Also, can you add a note about the metric `apiserver_encryption_config_controller_automatic_reload_last_timestamp_seconds ` that can be used to detect reload status?

File: content/en/docs/tasks/administer-cluster/kms-provider.md [hunk 1]
```diff
@@ -264,7 +264,7 @@ defined in a CustomResourceDefinition, your cluster must be running Kubernetes v
 
 1. Set the `--encryption-provider-config` flag on the kube-apiserver to point to the location of the configuration file.
 
-1. `--encryption-provider-config-automatic-reload` boolean argument determines if the file set by `--encryption-provider-config` should be automatically reloaded if the disk contents change. This enables key rotation without API server restarts.
+1. `--encryption-provider-config-automatic-reload` boolean argument determines if the file set by `--encryption-provider-config` should be automatically reloaded if the disk contents change. The Encryption configuration file is polled every minute to check if the disk content has changed. This enables key rotation without API server restarts. `apiserver_encryption_config_controller_automatic_reload_last_timestamp_seconds` metric correctly identifies when the new config becomes effective.
```
Comment (tengqm) [inline content/en/docs/tasks/administer-cluster/kms-provider.md:267]: Add a 'The' and wrap the long lines properly.
```suggestion
1. `--encryption-provider-config-automatic-reload` boolean argument determines
   if the file set by `--encryption-provider-config` should be automatically reloaded
   if the disk contents change. The Encryption configuration file is polled every minute
   to check if the disk content has changed. This enables key rotation without API
   server restarts.
   The `apiserver_encryption_config_controller_automatic_reload_last_timestamp_seconds`
   metric correctly identifies when the new config becomes effective.
```

File: content/en/docs/tasks/administer-cluster/kms-provider.md [hunk 2]
```diff
@@ -264,7 +264,13 @@ defined in a CustomResourceDefinition, your cluster must be running Kubernetes v
 
 1. Set the `--encryption-provider-config` flag on the kube-apiserver to point to the location of the configuration file.
 
-1. `--encryption-provider-config-automatic-reload` boolean argument determines if the file set by `--encryption-provider-config` should be automatically reloaded if the disk contents change. This enables key rotation without API server restarts.
+1. `--encryption-provider-config-automatic-reload` boolean argument determines
```
Comment (enj) [inline content/en/docs/tasks/administer-cluster/kms-provider.md:267]: I think the other page should have this detailed doc, and this KMS page should just link to that section.

File: content/en/docs/tasks/administer-cluster/kms-provider.md [hunk 3]
```diff
@@ -264,7 +264,13 @@ defined in a CustomResourceDefinition, your cluster must be running Kubernetes v
...
+   if the file set by `--encryption-provider-config` should be automatically reloaded
+   if the disk contents change. The Encryption configuration file is polled every minute
+   to check if the disk content has changed. This enables key rotation without API
+   server restarts.
+   The `apiserver_encryption_config_controller_automatic_reload_last_timestamp_seconds`
+   metric correctly identifies when the new config becomes effective.
```
Comment (enj) [inline content/en/docs/tasks/administer-cluster/kms-provider.md:273]:
```suggestion
   metric identifies when the new config becomes effective.
```

File: content/en/docs/tasks/administer-cluster/kms-provider.md [hunk 4]
```diff
@@ -264,7 +264,10 @@ defined in a CustomResourceDefinition, your cluster must be running Kubernetes v
 
 1. Set the `--encryption-provider-config` flag on the kube-apiserver to point to the location of the configuration file.
 
-1. `--encryption-provider-config-automatic-reload` boolean argument determines if the file set by `--encryption-provider-config` should be automatically reloaded if the disk contents change. This enables key rotation without API server restarts.
+1. `--encryption-provider-config-automatic-reload` boolean argument 
+   determines if the file set by `--encryption-provider-config` should be 
+   [automatically reloaded](/docs/tasks/administer-cluster/encrypt-data/#configure-automatic-reloading) if the disk contents change. This enables key 
+   rotation without API server restarts.
```
Comment (enj) [inline content/en/docs/tasks/administer-cluster/kms-provider.md:270]:
```suggestion
   [automatically reloaded](/docs/tasks/administer-cluster/encrypt-data/#configure-automatic-reloading) if the disk contents change.
```

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 5]
```diff
@@ -710,7 +710,11 @@ allows you to change the keys for encryption at rest without restarting the
 API server.
 
 To allow automatic reloading, configure the API server to run with:
-`--encryption-provider-config-automatic-reload=true`
+`--encryption-provider-config-automatic-reload=true`.
+When enabled, file changes are polled every minute to observe the modifications.
+The `apiserver_encryption_config_controller_automatic_reload_last_timestamp_seconds`
+metric identifies when the new config becomes effective.
```
Comment (enj) [inline content/en/docs/tasks/administer-cluster/encrypt-data.md:716]:
```suggestion
metric identifies when the new config becomes effective. This allows
encryption keys to be rotated without restarting the API server.
```

**Review by enj — CHANGES_REQUESTED** [review]

**Pull request merged**